### ISSUE TITLE
Root permission for apps

### ISSUE THREAD

**Issue by daryl0** [anbox/anbox#118]
I want install gapps from apk. For install it is necessary root permission.

With command adb shell and then SU I Take root permission but I stop it.

Help me.

**Comment by dispicio** [comment]
We need to modify android.img for this.
Extract the image (sudo to preserve permission/owner/groups):
`sudo unsquashfs android.img`
Then make your desired modifications (like cp su to /system/bin. Mind permissions/owner/groups of the new files).
Finally "repack" the image:
`mksquashfs squashfs-root android.img -b 131072 -comp xz -Xbcj x86`

Get the Google apps here (cp to /var/lib/anbox/rootfs/system/app): [http://opengapps.org/](http://opengapps.org/)
Latest SuperSU-zip (cp to /var/lib/anbox/rootfs/system/bin): [http://www.supersu.com/download](http://www.supersu.com/download)

I haven't been able to get GApps running though.. 😒
Maybe someone else is luckier?

**Comment by daryl0** [comment]
android image (android.img) I found in /snap/anbox/current ?

cd /snap/anbox/current and then you command? sudo unsquashfs android.img and then repack android image is ok?

**Comment by daryl0** [comment]
I try Extract android.img from  /snap/anbox/current but it is read only.

I download android.img from here (http://build.anbox.io/android-images/2017/04/12/android_1_amd64.img) 
Then i follow you command:
- sudo unsquashfs android.img
- modify permission in /system/bin (write/read only my user and other user only read)
- copy SU file (from previous download zip from here (https://s3-us-west-2.amazonaws.com/supersu/download/zip/SuperSU-v2.79-201612051815.zip) from directory x64 in /system/bin and libsupol.so under /system/lib64
- repack it with: mksquashfs squashfs-root android.img -b 131072 -comp xz -Xbcj x86
recompile anbox 

I try it and report here result.

**Comment by subjectdenied** [comment]
everytime i tried to change something in the unsquashed image, after squashing it again, the image is running for me. however adb won't see the emulator device anymore

**Comment by morphis** [comment]
@subjectdenied the ADB connection is a bit flawky. Something we should rework. Can you open a separate issue for that?

**Comment by dispicio** [comment]
Another try. Got play store starting, but as there is no network connection (another problem #51), it's pretty useless. I did not get root working! (tried to put su in /system/xbin with suid and symlink it from /system/bin). Anyhow, here's what I did for the play store:
1. `sudo unsquashfs android.img`
2. Download & extract [http://opengapps.org/](http://opengapps.org/)  (x86_64, 7.1, mini)
3. Extract Core/vending-all.tar.lz,  Core/gsflogin-all.tar.lz, Core/gsfcore-all.tar.lz, Core/gmscore-x86_64.tar.lz
4. Find and copy the folders Phonesky, GoogleLoginService, GoogleServicesFramework, PrebuiltGmsCore you extracted in 3. to squashfs-root/system/priv-app/
5. `sudo chown -R 100000:100000 Phonesky GoogleLoginService GoogleServicesFramework PrebuiltGmsCore` inside squashfs-root/system/priv-app
6. `sudo mksquashfs squashfs-root android.img -b 131072 -comp xz -Xbcj x86`

**Comment by subjectdenied** [comment]
@dispicio : session-manager is crashing for me after using the image modified as you explain, also the new image is 4x the size of the original one

**Comment by dispicio** [comment]
I guess the crashing is normal; this software is **pre-alpha**. You can try to fiddle around with the -comp compression parameters  in `mksquashfs` in order to make the image smaller (I get a roughly 15% bigger image than the original though, not 4x).

**Comment by subjectdenied** [comment]
@dispicio : size was my fault, crash is gone too. seems like the "device" is configured for tablets only. i see some new apps, also can see some i own

**Comment by daryl0** [comment]
@dispicio sorry I am exhausted. Kindly write me exactly all the steps, but especially the working directory in question as well as at the controls? On Linux I work with Android but I have a lot of practice.

Thanks and excuse me. :-(

**Comment by oleid** [comment]
@dispicio Thanks for letting me know how to get PlayStore working. 

FYI: I had a little trouble getting DNS working in the container (pinging was successfully tested via ```adb shell``` ). The anbox-bridge.sh script didn't work for me, however specifying a DNS server in ```/usr/lib/systemd/network/80-anbox-bridge.network``` successfully passed it down to the container.

The only problem now is that apps like _WhatsApp_ or _TomTom Go_  are not installable since _they are not available for my device_ -- according to PlayStore.

**Comment by asharptech** [comment]
@oleid install them from apkmirror.com ?

**Comment by oleid** [comment]
Thanks for the hint. It would seem, as if the TomTom app is only available for ARM. That's a pity. I guess one would have to hook up the runtime with qemu, as they did with the ASUS Zenphone.

**Comment by gilzamir** [comment]
After step 6,  where do i save generated android.img file?

**Comment by gilzamir** [comment]
@dispicio 
After step 6, where do i save generated android.img file?

**Comment by gilzamir** [comment]
dear @dispicio 

thanks a lot, your tip worked. Your tip worked. But I had to change the google service and play store in [settins / apps] permissions, granting these apps all permissions. I did not need to change the build.prop

> [Edit 2: Google Play Store launches, I'm able to login. There are no apps available though, now fiddling with system/build.prop.]

**Comment by markmcwong** [comment]
Hi I followed the step by dispicio and unsquashfs the android.img,
I also extracted those 4 folders in the correct place and i sudo chown them and squashing them again. 
Then I use `sudo anbox container-manager --android-image=/home/mark/anbox-work/anbox/android.img --data-path=/home/mark/anbox-data` and restart my computer.

I tried to use playstore but it doesnt work, any suggestion?
(Is su necessary in this case?)

**Comment by gilzamir** [comment]
Have you ever tried to change the permissions of google play store? Go in settings-->apps-->play store and choose permissions to play store.

**Comment by markmcwong** [comment]
I allow all permissions for play store and allows modify system settings but still no luck

**Comment by gilzamir** [comment]
You have to do the same with google play service.

**Comment by markmcwong** [comment]
@gilzamir it works and now youtube wont ask for google play framework and play store doesnt close after open but playstore just remain as a blank screen and when I clear data/cache  and restart it spinner appears for like 5s and disappear and remain blank screen

**Comment by subjectdenied** [comment]
@gilzamir : do you see all play store apps, or just a fraction of it? cause when i tried this days ago, i only saw about 4 apps available, that i didn't have installed yet on any of my devices before. i could get more apps shown by changing system/build.props, however never got the same pool as my phone would give me

**Comment by markmcwong** [comment]
@subjectdenied I cant see any play store apps as it remains a white blank screen.
I researched about the problem as people suggested to clear cache and data + remove google account

However when I tried to add my Google Account in Settings > Account it just close immediately

**Comment by gilzamir** [comment]
@subjectdenied @markmcwong I can see a few apps. It must be because the image android.img architecture is x86 and most of the apps just support arm. In addition, the build.prop file does not seem to be well configured. You can solve some of the problem by changing the build.prop file.

**Comment by gilzamir** [comment]
Dear, the internet is necessary for the play store works. Run the following command to internet to  work in the container

sudo /snap/anbox/current/bin/anbox-bridge.sh  start

**Comment by ktp420** [comment]
Anyone here build an arm android image and got arm apps running inside it?

**Comment by seanlano** [comment]
@ktp420 that would require an ARM CPU. Anbox doesn't emulate a CPU, it uses the host computer hardware.

**Comment by ktp420** [comment]
Yes I know but was just wondering if something tried it

**Comment by gilzamir** [comment]
@ktp420 It may be more helpful to try to get a image of the AndroidX86 project.  I think AndroidX86 has arm support, even though it's x86.

**Comment by ktp420** [comment]
I build the arm image for sources...just didn't want to download 40 gb if didn't need to.

**Comment by virajmalia** [comment]
@gilzamir 
Where did you save the squashed image?

**Comment by sjodiel** [comment]
@dispicio @gilzamir Hello, how did you guys get permission to record?
I am not able to copy the modified android.img, even by changing the permissions, the system reports: "read-only file system".

**Comment by sjodiel** [comment]
@dispicio @gilzamir Hello, how did you guys get permission to record?
I am not able to copy the modified android.img, even by changing the permissions, the system reports: "read-only file system".

**Comment by sjodiel** [comment]
How can I replace the android.img for my android.img modified?

**Comment by tbraeutigam** [comment]
Did anyone get the ARM binary support working?
I wasn't able to get houdini to work seeing how I couldn't register anything in binfmt in the container.
If someone managed -- do you have a guide? Or a working image to share?

**Comment by krzys-h** [comment]
The reason almost no apps appear in the play store is that Anbox by default doesn't report almost any features, not even the ones that are often declared implicitly like `android.hardware.touchscreen`. You can see what features the device reports as available using `adb shell pm list features`. I managed to get almost all apps to appear with the following changes:
```diff
diff -ruN squashfs-root/system/build.prop squashfs-root-modified/system/build.prop
--- squashfs-root/system/build.prop	2017-07-13 13:01:38.000000000 +0200
+++ squashfs-root-modified/system/build.prop	2017-07-26 23:32:34.517463493 +0200
@@ -26,8 +26,8 @@
 # ro.product.cpu.abi and ro.product.cpu.abi2 are obsolete,
 # use ro.product.cpu.abilist instead.
 ro.product.cpu.abi=x86_64
-ro.product.cpu.abilist=x86_64,x86
-ro.product.cpu.abilist32=x86
+ro.product.cpu.abilist=x86_64,x86,armeabi-v7a,armeabi
+ro.product.cpu.abilist32=x86,armeabi-v7a,armeabi
 ro.product.cpu.abilist64=x86_64
 ro.product.manufacturer=unknown
 ro.product.locale=en-US
@@ -71,3 +71,7 @@
 dalvik.vm.lockprof.threshold=500
 net.bt.name=Android
 dalvik.vm.stack-trace-file=/data/anr/traces.txt
+
+# Report GLES as available
+# 131072 ix 0x20000 in hex, which means GLES 2.0
+ro.opengles.version=131072
diff -ruN squashfs-root/system/etc/permissions/anbox.xml squashfs-root-modified/system/etc/permissions/anbox.xml
--- squashfs-root/system/etc/permissions/anbox.xml	2017-07-13 13:19:19.000000000 +0200
+++ squashfs-root-modified/system/etc/permissions/anbox.xml	2017-07-26 23:32:51.737464010 +0200
@@ -17,6 +17,19 @@
 <permissions>
     <feature name="android.software.freeform_window_management" />
     <feature name="android.hardware.ethernet" />
+
+	<feature name="android.hardware.touchscreen" />
+	<feature name="android.hardware.audio.output" />
+	<feature name="android.hardware.camera" />
+	<feature name="android.hardware.camera.any" />
+	<feature name="android.hardware.location" />
+	<feature name="android.hardware.location.gps" />
+	<feature name="android.hardware.location.network" />
+	<feature name="android.hardware.microphone" />
+	<feature name="android.hardware.screen.portrait" />
+	<feature name="android.hardware.screen.landscape" />
+	<feature name="android.hardware.wifi" />
+	<feature name="android.hardware.bluetooth" />
     
     <!-- Disallow device admin and profile owner on Anbox-->
     <unavailable-feature name="android.software.device_admin" />
@@ -25,8 +38,8 @@
     <!-- Disable support for hardware we don't support -->
     <unavailable-feature name="android.hardware.usb.accessory" />
     <unavailable-feature name="android.hardware.usb.host" />
-    <unavailable-feature name="android.hardware.wifi" />
-    <unavailable-feature name="android.hardware.bluetooth" />
+    <!--<unavailable-feature name="android.hardware.wifi" />-->
+    <!--<unavailable-feature name="android.hardware.bluetooth" />-->
     <unavailable-feature name="android.hardware.bluetooth_le" />
     <unavailable-feature name="android.hardware.sensor.compass" />
     <unavailable-feature name="android.hardware.sensor.accelerometer" />
```
I also made it display arm apps because I was trying to get libhoudini to work. Note that google play seems to cache this data somewhere, so every time you change these files and rebuild android.img you have to go to app settings and clear its data.

Unfortunately most apps I tried to run don't work because they either don't support `android.software.freeform_window_management` or - even worse - try to force the device into portrait mode.

**Comment by geeks-r-us** [comment]
Hey I made a little wrap up how to get PlayStore working.
It's a german post but google translate is your friend:
http://geeks-r-us.de/2017/08/26/android-apps-auf-dem-linux-desktop/

**Comment by jackwish** [comment]
@geeks-r-us wow, you integrated Houdini...

**Comment by morphis** [comment]
@geeks-r-us That is great! Can you create PRs to get the missing features in the .xml files added? See https://github.com/anbox/device_anbox/blob/master/anbox.xml for a place to do that

**Comment by geeks-r-us** [comment]
I can do this. But I do not know if this will work  or not haven't tested apps with bt for example...

**Comment by eugenio** [comment]
@geeks-r-us, I followed the procedure you are listing in the blog post above but after I resquash and put the image into place the anbox doesn't start anymore...

**Comment by step21** [comment]
Generally it works for me, but when logging in and the accepting the terms & conditions, it says to try again later. (after 2 factor authentication worked) Anybody know why or how to get some debug output?

**Comment by VibSel** [comment]
sudo anbox container-manager --data-path=~/anbox-data --android-image=~/anbox-work/android.img 
This command insn't working for me and the /snap/anbox/current/android.img file is read only for me.

**Comment by step21** [comment]
Unmount the snap. Unpack it with squashfs. Replace the android.img there with the new one. Pack it back together. You cannot write to snaps directly afaik. There is a blog post somewhere about it.

**Comment by VibSel** [comment]
How to unmount the snap folder? Normal unmount command is not working. It returns `mount: can't find /snap in /etc/fstab`

**Comment by step21** [comment]
It should work. It is not supposed to be in /etc/fstab Maybe make sure anbox is stopped and snap disabled first. Blog post http://geeks-r-us.de/2017/08/26/android-apps-auf-dem-linux-desktop/ (use google translate) Comments are especially important.

**Comment by VibSel** [comment]
Hey! @step21  I finally managed to complete this task. But now when I open Google Play Store it says, " Trouble connecting to server". The Internet doesn't seem to work in any app inside anbox.

**Comment by VibSel** [comment]
`anbox0    Link encap:Ethernet  HWaddr fe:5c:e4:cc:14:a7  
          inet addr:192.168.250.1  Bcast:0.0.0.0  Mask:255.255.255.0
          inet6 addr: fe80::24a2:7ff:fec1:efeb/64 Scope:Link
          UP BROADCAST RUNNING MULTICAST  MTU:1500  Metric:1
          RX packets:19 errors:0 dropped:0 overruns:0 frame:0
          TX packets:115 errors:0 dropped:0 overruns:0 carrier:0
          collisions:0 txqueuelen:1000 
          RX bytes:1256 (1.2 KB)  TX bytes:14774 (14.7 KB)
`

**Comment by geeks-r-us** [comment]
@VieSel
I updated the post with more bug fixing for connection issues.

And also added an initial version of an install script that adds ARM translation and Play Store to the Anbox image automatically

**Comment by lemrm** [comment]
@step21 
> Unmount the snap ... Replace the android.img there with the new one 
How to do this? I cant get it. @

**Comment by Rehazen** [comment]
@geeks-r-us i just literally start using linux 3 days ago, what must i do with script step by step in your blog? copy-paste in terminal?
something i did is copying your full script, and try to run in terminal, after that my anbox just in the "starting" screen (can't go in menu)

**Comment by datamaskinen** [comment]
it would be good if someone could put up a done .img file on github instead maybe

**Comment by shawe** [comment]
@step21 Your external link works for me. This page contains this script https://github.com/geeks-r-us/anbox-playstore-installer/raw/master/install-playstore.sh

On Antergos I need to replace line 106, with ```cp /var/lib/anbox/android.img .```
And from line 203 to the end with (this part seems to be very different):
```# update anbox snap images
#cd /var/lib/snapd/snaps
cp $WORKDIR/android.img /var/lib/anbox/android.img
$SUDO systemctl restart anbox-container-manager.service
```

Not works perfect, but is a step closer. Maybe as @morphis say, is needed a beeter https://github.com/anbox/device_anbox/blob/master/anbox.xml

**Comment by morphis** [comment]
For everyone interested, there is now a much easier way to modify the Android rootfs without touching the android.img. See https://github.com/anbox/anbox/pull/774 and https://docs.anbox.io/userguide/advanced/rootfs_overlay.html (wait with trying it out until the PR has landed)

**Comment by Shotman** [comment]
@morphis I've used the command listed on the official doc but I can't run any app as root, am I missing something ?
For instance I've sucessfuly installed the opengapps apk app, and ran anbox with the ``--rootfs=android-rootfs`` option but I can't check the "Use Root Permission" option in the app settings, I get a toast message saying : "Your device must be rooted in order to activate"

**Comment by LWJGLRobot** [comment]
it is easy to root anbox
1)unpack android.img
2)read and use this SuperSU.apk/META-INF/com/google/android/update-binary line:85
or do this (approximately same as update-binary script at lines 85-114):
cp common/Superuser.apk system/app/SuperSU/SuperSU.apk
cp x64/su  system/xbin/su
cp x64/su  system/bin/.ext/.su
cp x64/su  system/xbin/daemonsu  
cp x64/supolicy system/xbin/supolicy
cp x64/libsupol.so system/lib64/libsupol.so
cp system/bin/app_process64 system/bin/app_process_init
cp system/bin/app_process64 system/bin/app_process64_original
cp system/xbin/daemonsu system/bin/app_process 
cp system/xbin/daemonsu system/bin/app_process64
chmod +x <all-files-you-changed>
3)pack android.img and put it to anbox folder, where from it taken.
4)restart container-manager
5)restart session-manager

i got supersu work.
![screenshot from 2018-07-16 19-21-11](https://user-images.githubusercontent.com/25325598/42770478-79b4affe-892d-11e8-85fb-594ce49f421f.png)

**Comment by eMPee584** [comment]
Ok @morphis already commented, that should be easily achievable with the `--use-rootfs-overlay` option... however, I was unable to make supersu work properly.
The overlaid /system/xbin/su just doesn't mask the existing one, which lacks exec permissions.

How is this, why are only those files in `rootfs-overlay` not already in `rootfs` visible in `combined-rootfs` ?
Here's my attempt at overlaying SuperSU v2.82 ...
[rootfs-overlay.tar.gz](https://github.com/anbox/anbox/files/2198788/rootfs-overlay.tar.gz)

**Comment by morphis** [comment]
@eMPee584 That got fix with one of the latest PRs. See https://github.com/anbox/anbox/pull/811

**Comment by eMPee584** [comment]
Wohoo! Seems to work now, with the files I posted earlier placed in the rootfs overlay. :cowboy_hat_face:

**Comment by oxwivi** [comment]
I'm on Debian unstable and I got Anbox running by `sudo apt install anbox`, no PPA or drivers, and that's great. But there's no F-Droid, according to this thread, I need to fiddle with some script in the `snap` directory. Any idea where I can find those in the Debian repo's package?

**Comment by oxwivi** [comment]
I went through the thread again, and I realized I've no idea what I'm talking about. I'll go through this all tomorrow, good night.

**Comment by eMPee584** [comment]
@oxwivi the gapps thing is quite a mission. I actually got it the play store & maps f.e. to work with https://github.com/geeks-r-us/anbox-playstore-installer/pull/14 , albeit with a "slight" rendering issue...
![image](https://user-images.githubusercontent.com/1227833/46574888-2c064880-c9ab-11e8-9d05-10785418ed53.png)

**Comment by oxwivi** [comment]
> @oxwivi the gapps thing is quite a mission, infusing the houdini libs into the local image and such... I actually got the play store & f.e. maps to work with [geeks-r-us/anbox-playstore-installer#14](https://github.com/geeks-r-us/anbox-playstore-installer/pull/14) , albeit with a "slight" rendering issue (using proprietary nvidia x11 driver)...
> ![image](https://user-images.githubusercontent.com/1227833/46574888-2c064880-c9ab-11e8-9d05-10785418ed53.png)

I've a couple of questions:
 - with the premise it's not a `snap` install, why doesn't it check for `android.img` at `/var/lib/anbox/android.img` or attempting to download from `https://build.anbox.io/android-images` as specified in the `file:///usr/share/doc/anbox/README.Debian`?
 - Is there anything I can do to the script to let it install F-Droid in it?

The GApps scripts aside, why is there no way to just plop in an `.apk` into the root of the container where we can navigate to using Files and installing it like a normal device?

**Comment by oxwivi** [comment]
![screenshot_20181008_135652](https://user-images.githubusercontent.com/15462036/46602740-135c7680-cb02-11e8-801b-f3307675256b.png)
I used @eMPee584's script, got Anbox running, logged in with my credentials, accepted terms and conditions and was greeted with this. What did you do to deal with this? I think it's to do with the lacking device specifications some of the comments mentioned.

**Comment by oxwivi** [comment]
The failure was because of permissions, I didn't give permissions to Google Play Store and Services. After that, it's logging in just fine, but the problem is it crashes before I can do anything. At some times, opening the menu on the Store will freeze it, others, I was able to go to My Apps and Games where the loading animation ran perpetually until I interacted with it and it crashed. Scrolling down for a few seconds also crashes Play Store.

**Comment by eMPee584** [comment]
`adb logcat -sv color \*:E`

**Comment by oxwivi** [comment]
> `adb logcat -sv color \*:E`

I didn't want to install `adb`, but now I went ahead and did it. However, this command causes a veritable ejaculation on my terminal. With Anbox session running but no app, not even the manager is running, the constant output is already making it impossible to follow, let alone anything specific due to my actions.

**Comment by ShapeShifter499** [comment]
@eMPee584 Do you think your overlay supersu could be modified for Magisk? This would possibly be good when running apps that don't like root or other non-approved system properties.

**Comment by eMPee584** [comment]
> @eMPee584 Do you think your overlay supersu could be modified for Magisk? This would possibly be good when running apps that don't like root or other non-approved system properties.

Magisk has its own su.

**Comment by ShapeShifter499** [comment]
@eMPee584 I tried poking around to getting it installed but I don't know which file to patch since there is no bootloader.  I'm guessing that a img file for Magisk needs to be put at /data/magisk.img inside of the installed Android image. Then Anbox needs to be pointed at the one of the included magisk binaries to be loaded when it starts. But I wasn't 100% sure how to do this.

**Comment by gggauravgandhi** [comment]
> Ok @morphis already commented, that should be easily achievable with the `--use-rootfs-overlay` option... however, I was unable to make supersu work properly.
> The overlaid /system/xbin/su just doesn't mask the existing one, which lacks exec permissions.
> 
> How is this, why are only those files in `rootfs-overlay` not already in `rootfs` visible in `combined-rootfs` ?
> Here's my attempt at overlaying SuperSU v2.82 ...
> [rootfs-overlay.tar.gz](https://github.com/anbox/anbox/files/2198788/rootfs-overlay.tar.gz)

This works now, thanks.

**Comment by baptx** [comment]
@dispicio here is the file /system/build.prop that is used by Android x86 in VirtualBox, where I have access to popular apps like Firefox which are not available with Anbox by default:
`
# begin build properties
# autogenerated by buildinfo.sh
ro.build.id=N2G48H
ro.build.display.id=android_x86_64-userdebug 7.1.2 N2G48H eng.cwhuan.20180502.135503 test-keys
ro.build.version.incremental=eng.cwhuan.20180502.135503
ro.build.version.sdk=25
ro.build.version.preview_sdk=0
ro.build.version.codename=REL
ro.build.version.all_codenames=REL
ro.build.version.release=7.1.2
ro.build.version.security_patch=2017-10-05
ro.build.version.base_os=
ro.build.date=Wed May  2 13:55:03 CST 2018
ro.build.date.utc=1525240503
ro.build.type=userdebug
ro.build.user=cwhuang
ro.build.host=ws.cwhuang.info
ro.build.tags=test-keys
ro.build.flavor=android_x86_64-userdebug
ro.product.model=Generic Android-x86_64
ro.product.brand=Android-x86
ro.product.name=android_x86_64
ro.product.device=x86_64
ro.product.board=
# ro.product.cpu.abi and ro.product.cpu.abi2 are obsolete,
# use ro.product.cpu.abilist instead.
ro.product.cpu.abi=x86_64
ro.product.cpu.abilist=x86_64,x86,arm64-v8a,armeabi-v7a,armeabi
ro.product.cpu.abilist32=x86,armeabi-v7a,armeabi
ro.product.cpu.abilist64=x86_64,arm64-v8a
ro.product.manufacturer=unknown
ro.product.locale=en-US
ro.wifi.channels=
ro.board.platform=android-x86
# ro.build.product is obsolete; use ro.product.device
ro.build.product=x86_64
# Do not try to parse description, fingerprint, or thumbprint
ro.build.description=android_x86_64-userdebug 7.1.2 N2G48H eng.cwhuan.20180502.135503 test-keys
ro.build.fingerprint=Android-x86/android_x86_64/x86_64:7.1.2/N2G48H/cwhuan05021355:userdebug/test-keys
ro.build.characteristics=tablet
# end build properties

#
# ADDITIONAL_BUILD_PROPERTIES
#
ro.com.android.dateformat=MM-dd-yyyy
ro.ril.hsxpa=1
ro.ril.gprsclass=10
keyguard.no_require_sim=true
ro.com.android.dataroaming=true
media.sf.hwaccel=1
media.sf.omx-plugin=libffmpeg_omx.so
media.sf.extractor-plugin=libffmpeg_extractor.so
ro.opengles.version=196608
dalvik.vm.heapstartsize=16m
dalvik.vm.heapgrowthlimit=192m
dalvik.vm.heapsize=512m
dalvik.vm.heaptargetutilization=0.75
dalvik.vm.heapminfree=512k
dalvik.vm.heapmaxfree=8m
ro.com.google.gmsversion=7.1_r2
ro.com.google.clientidbase=android-asus
ro.com.google.clientidbase.ms=android-asus
ro.com.google.clientidbase.am=android-asus
ro.com.google.clientidbase.gmm=android-asus
ro.com.google.clientidbase.yt=android-asus
ro.setupwizard.mode=ENABLED
ro.dalvik.vm.isa.arm=x86
ro.enable.native.bridge.exec=1
ro.dalvik.vm.isa.arm64=x86_64
ro.enable.native.bridge.exec64=1
ro.carrier=unknown
ro.config.notification_sound=OnTheHunt.ogg
ro.config.alarm_alert=Alarm_Classic.ogg
persist.sys.dalvik.vm.lib.2=libart.so
dalvik.vm.isa.x86_64.variant=x86_64
dalvik.vm.isa.x86_64.features=default
dalvik.vm.isa.x86.variant=x86
dalvik.vm.isa.x86.features=default
dalvik.vm.lockprof.threshold=500
net.bt.name=Android
dalvik.vm.stack-trace-file=/data/anr/traces.txt
`

**Comment by DerTopher** [comment]
> We need to modify android.img for this.
> Extract the image (sudo to preserve permission/owner/groups):
> `sudo unsquashfs android.img`
> Then make your desired modifications (like cp su to /system/bin. Mind permissions/owner/groups of the new files).
> Finally "repack" the image:
> `mksquashfs squashfs-root android.img -b 131072 -comp xz -Xbcj x86`
> 
> Get the Google apps here (cp to /var/lib/anbox/rootfs/system/app): http://opengapps.org/
> Latest SuperSU-zip (cp to /var/lib/anbox/rootfs/system/bin): http://www.supersu.com/download
> 
> I haven't been able to get GApps running though.. unamused
> Maybe someone else is luckier?

I used your hints for resolving a different issue for me that concerned the keymapping in anbox. [See here](https://github.com/anbox/anbox/issues/253)

For anyone trying this solution note that you need to assing a different name to the image you are creating when respacking the image
i.e.
`mksquashfs squashfs-root android-new.img -b 131072 -comp xz -Xbcj x86`

then you can remove the original image and rename the repacked one to android.img
`rm android.img`
`mv android-new.img android.img`

Without doing that the modified files will be added under a different name to the existing img.

Note that you will need to restart for the changes to take effect.

**Comment by fsevenm** [comment]
> it is easy to root anbox
> 1)unpack android.img
> 2)read and use this SuperSU.apk/META-INF/com/google/android/update-binary line:85
> or do this (approximately same as update-binary script at lines 85-114):
> cp common/Superuser.apk system/app/SuperSU/SuperSU.apk
> cp x64/su system/xbin/su
> cp x64/su system/bin/.ext/.su
> cp x64/su system/xbin/daemonsu
> cp x64/supolicy system/xbin/supolicy
> cp x64/libsupol.so system/lib64/libsupol.so
> cp system/bin/app_process64 system/bin/app_process_init
> cp system/bin/app_process64 system/bin/app_process64_original
> cp system/xbin/daemonsu system/bin/app_process
> cp system/xbin/daemonsu system/bin/app_process64
> chmod +x all-files-you-changed
> 3)pack android.img and put it to anbox folder, where from it taken.
> 4)restart container-manager
> 5)restart session-manager
> 
> i got supersu work.
> ![screenshot from 2018-07-16 19-21-11](https://user-images.githubusercontent.com/25325598/42770478-79b4affe-892d-11e8-85fb-594ce49f421f.png)

Sorry, bring this to 2021. But I wonder how can you ` 3)pack android.img and put it to anbox folder, where from it taken.`. Where `/snap/anbox/186/android.img` and `/snap/anbox/current/android.img` is read only file system. I cannot remove or replace it even with `sudo`?

**Comment by baptx** [comment]
@fsevenm I shared detailed steps on https://github.com/anbox/anbox/issues/1368#issuecomment-622564745 in case it helps.

**Comment by fsevenm** [comment]
Thanks @baptx. Btw at the end, I replace anbox default image with the one with supersu, then I rebuilt it myself.

**Comment by taghiahmadi02** [comment]
> Another try. Got play store starting, but as there is no network connection (another problem #51), it's pretty useless. I did not get root working! (tried to put su in /system/xbin with suid and symlink it from /system/bin). Anyhow, here's what I did for the play store:
> 
> 1. `sudo unsquashfs android.img`
> 2. Download & extract http://opengapps.org/  (x86_64, 7.1, mini)
> 3. Extract Core/vending-all.tar.lz,  Core/gsflogin-all.tar.lz, Core/gsfcore-all.tar.lz, Core/gmscore-x86_64.tar.lz
> 4. Find and copy the folders Phonesky, GoogleLoginService, GoogleServicesFramework, PrebuiltGmsCore you extracted in 3. to squashfs-root/system/priv-app/
> 5. `sudo chown -R 100000:100000 Phonesky GoogleLoginService GoogleServicesFramework PrebuiltGmsCore` inside squashfs-root/system/priv-app
> 6. `sudo mksquashfs squashfs-root android.img -b 131072 -comp xz -Xbcj x86`
> 
> [Edit: Got network running after installing dnsmasq and getting anbox-bridge.sh from this repo. `sudo ./anbox-bridge.sh start`. On Arch Linux] [Edit 2: Google Play Store launches, I'm able to login. There are no apps available though, now fiddling with system/build.prop.] [Edit 3: Has anyone a x86 chromebook and is willing to provide its build.prop?]
